File: TI2.1  SOAP_JAVA_EUREKA_BANK_GR08/01.SERVIDOR/ws_eureka_bank_soap_java/src/main/java/ec/edu/monster/Controlador/CuentaDAO.java
package ec.edu.monster.Controlador;


import ec.edu.monster.db.AccesoBD;
import ec.edu.monster.modelo.CuentaModel;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

public class CuentaDAO {

    // Actualizar el saldo de la cuenta cuando se hace un Movimiento (Retiro, Deposito, Transferencia)
    public boolean actualizarSaldoCuenta(String codigoCuenta, String valorMovimiento) throws SQLException {
        String sql = "UPDATE cuenta "
                + "SET dec_cuensaldo = dec_cuensaldo + ?, "
                + "int_cuencontmov = int_cuencontmov + 1 "
                + "WHERE chr_cuencodigo = ?";
        try (Connection connection = AccesoBD.getConnection(); PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.setDouble(1, Double.parseDouble(valorMovimiento)); // El valor que deseas sumar (positivo o negativo)
            statement.setString(2, codigoCuenta);   // El código de la cuenta a actualizar

            int rowsUpdated = statement.executeUpdate(); // Ejecuta la actualización
            return rowsUpdated > 0; // Retorna true si se actualizó al menos una fila
        }
    }

    // Método para obtener una cuenta por su Código
    public CuentaModel obtenerCuentaPorCodigo(String codigoCuenta) throws SQLException {
        String sql = "SELECT * FROM cuenta WHERE chr_cuencodigo = ?";
        try (Connection connection = AccesoBD.getConnection(); PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.setString(1, codigoCuenta);
            var resultSet = statement.executeQuery();
            if (resultSet.next()) {
                CuentaModel cuenta = new CuentaModel();
                cuenta.setChrCuenCodigo(resultSet.getString("chr_cuencodigo"));
                cuenta.setChrMoneCodigo(resultSet.getString("chr_monecodigo"));
                cuenta.setChrSucucodigo(resultSet.getString("chr_sucucodigo"));
                cuenta.setChrEmplCreaCuenta(resultSet.getString("chr_emplcreacuenta"));
                cuenta.setChrClieCodigo(resultSet.getString("chr_cliecodigo"));
                cuenta.setDecCuenSaldo(resultSet.getDouble("dec_cuensaldo"));
                cuenta.setDttCuenFechaCreacion(resultSet.getDate("dtt_cuenfechacreacion"));
                cuenta.setVchCuenEstado(resultSet.getString("vch_cuenestado"));
                cuenta.setIntCuenContMov(resultSet.getInt("int_cuencontmov"));
                cuenta.setChrCuenClave(resultSet.getString("chr_cuenclave"));
                return cuenta;
            }
        }
        return null;
    }
}

File: Tl2.4 RESTFULL_DOTNET_EUREKA_BANK_GR08/01.SERVIDOR/ws_eureka_bank_restful_dotnet/ec.edu.monster.dao/CuentaDAO.cs
using ec.edu.monster.db;
using ec.edu.monster.model;
using Microsoft.Data.SqlClient;

namespace ec.edu.monster.dao;

public class CuentaDAO
{
    private readonly IConfiguration _configuration;

    public CuentaDAO(IConfiguration configuration)
    {
        _configuration = configuration;
    }

    // Actualizar el saldo de la cuenta cuando se hace un Movimiento (Retiro, Deposito, Transferencia)
    public bool ActualizarSaldoCuenta(string codigoCuenta, string valorMovimiento)
    {
        string sql = "UPDATE cuenta " +
                    "SET dec_cuensaldo = dec_cuensaldo + @valor, " +
                    "int_cuencontmov = int_cuencontmov + 1 " +
                    "WHERE chr_cuencodigo = @codigo";
        
        using (var connection = AccesoBD.GetConnection(_configuration))
        {
            connection.Open();
            using (var command = new SqlCommand(sql, connection))
            {
                command.Parameters.AddWithValue("@valor", double.Parse(valorMovimiento));
                command.Parameters.AddWithValue("@codigo", codigoCuenta);

                int rowsUpdated = command.ExecuteNonQuery();
                return rowsUpdated > 0;
            }
        }
    }

    // Método para obtener una cuenta por su Código
    public CuentaModel? ObtenerCuentaPorCodigo(string codigoCuenta)
    {
        string sql = "SELECT * FROM cuenta WHERE chr_cuencodigo = @codigo";
        
        using (var connection = AccesoBD.GetConnection(_configuration))
        {
            connection.Open();
            using (var command = new SqlCommand(sql, connection))
            {
                command.Parameters.AddWithValue("@codigo", codigoCuenta);
                
                using (var reader = command.ExecuteReader())
                {
                    if (reader.Read())
                    {
                        int idxCuenCodigo = reader.GetOrdinal("chr_cuencodigo");
                        int idxMoneCodigo = reader.GetOrdinal("chr_monecodigo");
                        int idxSucucodigo = reader.GetOrdinal("chr_sucucodigo");
                        int idxEmplCreaCuenta = reader.GetOrdinal("chr_emplcreacuenta");
                        int idxClieCodigo = reader.GetOrdinal("chr_cliecodigo");
                        int idxCuenSaldo = reader.GetOrdinal("dec_cuensaldo");
                        int idxCuenFechaCreacion = reader.GetOrdinal("dtt_cuenfechacreacion");
                        int idxCuenEstado = reader.GetOrdinal("vch_cuenestado");
                        int idxCuenContMov = reader.GetOrdinal("int_cuencontmov");
                        int idxCuenClave = reader.GetOrdinal("chr_cuenclave");

                        var cuenta = new CuentaModel
                        {
                            ChrCuenCodigo = reader.GetString(idxCuenCodigo),
                            ChrMoneCodigo = reader.IsDBNull(idxMoneCodigo) ? null : reader.GetString(idxMoneCodigo),
                            ChrSucucodigo = reader.IsDBNull(idxSucucodigo) ? null : reader.GetString(idxSucucodigo),
                            ChrEmplCreaCuenta = reader.IsDBNull(idxEmplCreaCuenta) ? null : reader.GetString(idxEmplCreaCuenta),
                            ChrClieCodigo = reader.IsDBNull(idxClieCodigo) ? null : reader.GetString(idxClieCodigo),
                            DecCuenSaldo = Convert.ToDouble(reader.GetDecimal(idxCuenSaldo)),
                            DttCuenFechaCreacion = reader.IsDBNull(idxCuenFechaCreacion) ? null : reader.GetDateTime(idxCuenFechaCreacion),
                            VchCuenEstado = reader.IsDBNull(idxCuenEstado) ? null : reader.GetString(idxCuenEstado),
                            IntCuenContMov = reader.GetInt32(idxCuenContMov),
                            ChrCuenClave = reader.IsDBNull(idxCuenClave) ? null : reader.GetString(idxCuenClave)
                        };
                        return cuenta;
                    }
                }
            }
        }
        return null;
    }
}

